### ISSUE TITLE
Container fails to start due to permission issue when removing temp file in /output or /watch

### ISSUE THREAD

**Issue by aurrak** [jlesage/docker-handbrake#189]
In `docker-handbrake/rootfs/etc/cont-init.d/handbrake.sh`, `mktemp` was successful on creating a temp file in the output/watch folders (I verified the temp file was there in the filesystem), but `rm $TMPFILE` has issue removing it due to permission issue, causing the container failing to start.

```bash
# Take ownership of the output directory.
for i in $(seq 1 ${AUTOMATED_CONVERSION_MAX_WATCH_FOLDERS:-5}); do
    eval "DIR=\"\${AUTOMATED_CONVERSION_OUTPUT_DIR_$i:-/output}\""

    if [ ! -d "$DIR" ]; then
        log "ERROR: Output folder '$DIR' doesn't exist."
        exit 1
    elif ! chown $USER_ID:$GROUP_ID "$DIR"; then
        # Failed to take ownership of /output.  This could happen when,
        # for example, the folder is mapped to a network share.
        # Continue if we have write permission, else fail.
        TMPFILE="$(s6-setuidgid $USER_ID:$GROUP_ID mktemp "$DIR"/.test_XXXXXX 2>/dev/null)"
        if [ $? -eq 0 ]; then
            # Success, we were able to write file.
            rm "$TMPFILE"
        else
            log "ERROR: Failed to take ownership and no write permission on '$DIR'."
            exit 1
        fi
    fi
done
```

See full output log below:

```
handbrake    | [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
handbrake    | [s6-init] ensuring user provided files have correct perms...exited 0.
handbrake    | [fix-attrs.d] applying ownership & permissions fixes...
handbrake    | [fix-attrs.d] done.
handbrake    | [cont-init.d] executing container initialization scripts...
handbrake    | [cont-init.d] 00-app-niceness.sh: executing...
handbrake    | [cont-init.d] 00-app-niceness.sh: exited 0.
handbrake    | [cont-init.d] 00-app-script.sh: executing...
handbrake    | [cont-init.d] 00-app-script.sh: exited 0.
handbrake    | [cont-init.d] 00-app-user-map.sh: executing...
handbrake    | [cont-init.d] 00-app-user-map.sh: exited 0.
handbrake    | [cont-init.d] 00-clean-logmonitor-states.sh: executing...
handbrake    | [cont-init.d] 00-clean-logmonitor-states.sh: exited 0.
handbrake    | [cont-init.d] 00-clean-tmp-dir.sh: executing...
handbrake    | [cont-init.d] 00-clean-tmp-dir.sh: exited 0.
handbrake    | [cont-init.d] 00-set-app-deps.sh: executing...
handbrake    | [cont-init.d] 00-set-app-deps.sh: exited 0.
handbrake    | [cont-init.d] 00-set-home.sh: executing...
handbrake    | [cont-init.d] 00-set-home.sh: exited 0.
handbrake    | [cont-init.d] 00-take-config-ownership.sh: executing...
handbrake    | [cont-init.d] 00-take-config-ownership.sh: exited 0.
handbrake    | [cont-init.d] 00-xdg-runtime-dir.sh: executing...
handbrake    | [cont-init.d] 00-xdg-runtime-dir.sh: exited 0.
handbrake    | [cont-init.d] 10-certs.sh: executing...
handbrake    | [cont-init.d] 10-certs.sh: exited 0.
handbrake    | [cont-init.d] 10-cjk-font.sh: executing...
handbrake    | [cont-init.d] 10-cjk-font.sh: installing CJK font...
handbrake    | [cont-init.d] 10-cjk-font.sh: fetch http://dl-cdn.alpinelinux.org/alpine/edge/testing/x86_64/APKINDEX.tar.gz
handbrake    | [cont-init.d] 10-cjk-font.sh: fetch http://dl-cdn.alpinelinux.org/alpine/v3.12/main/x86_64/APKINDEX.tar.gz
handbrake    | [cont-init.d] 10-cjk-font.sh: fetch http://dl-cdn.alpinelinux.org/alpine/v3.12/community/x86_64/APKINDEX.tar.gz
handbrake    | [cont-init.d] 10-cjk-font.sh: (1/1) Installing wqy-zenhei (0.9.47-r0)
handbrake    | [cont-init.d] 10-cjk-font.sh: Executing fontconfig-2.13.1-r2.trigger
handbrake    | [cont-init.d] 10-cjk-font.sh: Executing mkfontscale-1.2.1-r1.trigger
handbrake    | [cont-init.d] 10-cjk-font.sh: OK: 168 MiB in 158 packages
handbrake    | [cont-init.d] 10-cjk-font.sh: exited 0.
handbrake    | [cont-init.d] 10-nginx.sh: executing...
handbrake    | [cont-init.d] 10-nginx.sh: exited 0.
handbrake    | [cont-init.d] 10-vnc-password.sh: executing...
handbrake    | stored passwd in file: /root/.vncpass
handbrake    | [cont-init.d] 10-vnc-password.sh: exited 0.
handbrake    | [cont-init.d] 10-web-index.sh: executing...
handbrake    | [cont-init.d] 10-web-index.sh: exited 0.
handbrake    | [cont-init.d] 95-check-optical-drive.sh: executing...
handbrake    | [cont-init.d] 95-check-optical-drive.sh: looking for usable optical drives...
handbrake    | [cont-init.d] 95-check-optical-drive.sh: no usable optical drive found.
handbrake    | [cont-init.d] 95-check-optical-drive.sh: exited 0.
handbrake    | [cont-init.d] 95-check-qsv.sh: executing...
handbrake    | [cont-init.d] 95-check-qsv.sh: Processor: Intel(R) Core(TM) i5-8600 CPU @ 3.10GHz
handbrake    | [cont-init.d] 95-check-qsv.sh: exited 0.
handbrake    | [cont-init.d] handbrake.sh: executing...
handbrake    | [cont-init.d] handbrake.sh: generating machine-id...
handbrake    | [cont-init.d] handbrake.sh: core dump file location: core
handbrake    | [cont-init.d] handbrake.sh: core dump file size: unlimited (blocks)
handbrake    | chown: changing ownership of '/output': Operation not permitted
handbrake    | rm: cannot remove '/output/.test_516U4c': Permission denied        <----------
handbrake    | [cont-init.d] handbrake.sh: exited 1.
handbrake    | [services.d] stopping services
handbrake    | [services.d] stopping s6-fdholderd...
handbrake    | [cont-finish.d] executing container finish scripts...
handbrake    | [cont-finish.d] done.
handbrake    | [s6-finish] syncing disks.
handbrake    | [s6-finish] sending all processes the TERM signal.
handbrake    | [s6-finish] sending all processes the KILL signal and exiting.
handbrake exited with code 1
```

Shouldn't `rm "$TMPFILE"` be `s6-setuidgid $USER_ID:$GROUP_ID  rm "$TMPFILE"` instead?

**Comment by aurrak** [comment]
I confirmed that changing `rm "$TMPFILE"` to `s6-setuidgid $USER_ID:$GROUP_ID rm "$TMPFILE"` in `handbrake.sh` resolved my permission issue and the container's able to start successfully.

**Comment by aurrak** [comment]
Looks like this was resolved by commit `dfcf86624d6e4240f3bad2b7f673dca33efa1d70`. Thanks!
